### PR TITLE
Refactor/lazy load

### PR DESCRIPTION
**Pull request by maoznir**
Refactor of advanced images utils: lazy load, placeholder, accessibility.

### CODE REVIEW

File: test/spec/util-spec.js [hunk 1]
```diff
@@ -53,4 +53,25 @@ describe("util", function () {
...
+    });
+
+    it("should call onIntersect", function () {
+      const {detectIntersection} = cloudinary.Util;
+
+      setTimeout(function() {
+        detectIntersection({}, onIntersect);
+      }, 100);
+
+      expect(onIntersect).not.toHaveBeenCalled();
```
Comment (patrick-tolosa) [inline test/spec/util-spec.js:70]: Why do we need the timer here?
Can't you call expect before and after you run detectIntersection?

File: test/spec/lazy-load-core-spec.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    };
+    it('should lazy load image on scroll', function (done) {
+      var lazy1 = testDocument.getElementById('lazy1');
+      var url = lazy1.getAttribute('data-src');
+      expect(url.startsWith('http')).toEqual(true);
+
+      cloudinary.Util.detectIntersection(lazy1, lazyLoad(lazy1));
+
+      setTimeout(()=>{
+        expect(lazy1.getAttribute('src')).toEqual(null);
```
[the remark this answers is not in the document]
Comment (maoznir) [reply test/spec/lazy-load-core-spec.js:52]: I'll change the ids and variable names to be more meaningfull

File: test/spec/util-spec.js (hunk 1 above)
Comment (maoznir) [reply test/spec/util-spec.js:70]: Simplify the test and state that it checks worst case scenario and fallsback to call the callback

File: src/transformation.js [hunk 3]
```diff
@@ -560,6 +561,11 @@ class Transformation extends TransformationBase {
   /*
     Transformation Parameters
   */
+  accessibility(value) {
+    this.chain();
+    return this.effect(ACCESSIBILITY_MODES[value]);
```
Comment (asisayag2) [inline src/transformation.js:566]: It's currently only effects, what if tomorrow we will have something else?

File: src/transformation.js [hunk 4]
```diff
@@ -778,6 +784,19 @@ class Transformation extends TransformationBase {
     return this.param(value, "page", "pg");
   }
 
+  placeholder(value) {
+    const mode = PLACEHOLDER_IMAGE_MODES[value] || PLACEHOLDER_IMAGE_MODES.blur;
+    const width = this.getValue("width");
+    const height = this.getValue("height");
+    let transformations = mode({width, height});
```
Comment (asisayag2) [inline src/transformation.js:791]: Only predominant accepts width and height and according to the spec and the ref implementation, it doesn't have to. 
If you do, we do the manipulation on a single pixel. If not, we do another calc. 

In addition, the entire code that go to consts file and the returned value is a method is strange. 
it took me 7 minutes to understand what's going on here.

File: src/transformation.js [hunk 5]
```diff
@@ -778,6 +784,19 @@ class Transformation extends TransformationBase {
...
+    this.chain();
+    transformations.forEach(t=>{
+      this.rawTransformation(new Transformation(t));
+      this.chain();
+    });
```
Comment (asisayag2) [inline src/transformation.js:796]: Pretty sure that `Transofmration(PLACEHOLDER_IMAGE_MODES["pixelate"])` will work

File: src/cloudinary.js [hunk 6]
```diff
@@ -206,6 +207,23 @@ class Cloudinary {
...
 
+  /**
+   * Generates a URL for an image intended to be used as a placeholder for the specified image.
+   * @param {string} publicId - original image public id
+   * @param {string} placeholderType - type of placeholder: 'blur'/'pixelate'/'predominant-color'/'vectorize'
+   * @param {Object} options - The {@link Transformation} parameters to include in the URL.
+   * @return {string}
+   */
+  placeholder_url(publicId, placeholderType = "blur", options = {}) {
+    const {config, transformation} = splitOptions(options);
```
[the remark this answers is not in the document]
Comment (maoznir) [reply src/cloudinary.js]: removed this splitting logic and moved placeholder & accessibility to be handled by cloudinary.utl()

File: src/transformation.js (hunk 3 above)
Comment (maoznir) [reply src/transformation.js]: this is now handled by cloudinary.url() and will be updated as needed

File: src/transformation.js (hunk 4 above)
Comment (maoznir) [reply src/transformation.js]: Converted from functions to just arrays and moved the logic to cloudinary.url()

File: src/transformation.js (hunk 5 above)
Comment (maoznir) [reply src/transformation.js]: you're right, fixed

File: test/spec/lazy-load-core-spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (maoznir) [reply test/spec/lazy-load-core-spec.js]: @patrick-tolosa I update this as we said and also reduced duplication by moving the shared test to lazyLoadTestBase.js

File: test/spec/lazy-load-jquery-spec.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
+let describeLazyLoadTest = describe;
+
```
[the remark this answers is not in the document]
Comment (maoznir) [reply test/spec/lazy-load-jquery-spec.js]: Moved to a single shared file lazyLoadTestBase.js

**Comment by maoznir** [comment]
was done in https://github.com/cloudinary/cloudinary_js/commit/b3dd7b6e71f4c0aa4ec4190d3356acc475f87d22

**Pull request closed without merging**